pkg/cri/store/image/image.go

```diff
@@ -212,7 +212,7 @@ func (s *store) add(img Image) error {
 		return nil
 	}
 	// Or else, merge and sort the references.
-	i.References = docker.Sort(util.MergeStringSlices(i.References, img.References))
+	i.References = docker.StrictSort(util.MergeStringSlices(i.References, img.References))
```

Comment:
Now the image store doesn't modify the image ref, which fixes an inconsistency between the `refCache` and `image.References`.

Although inconsistent images are already ignored in the `cri plugin service`, I think it's still necessary to fix the bug in the image store.